<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1038</width>
    <height>921</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="HLayout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="sizeConstraint">
         <enum>QLayout::SetFixedSize</enum>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Current Node:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="currentNode"/>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBox_drawCurrent">
          <property name="text">
           <string>Draw Current</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line_4">
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="lineWidth">
           <number>4</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Opération:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="Operation">
          <item>
           <property name="text">
            <string>Union</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Intersection</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Différence</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Id fils gauche</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="id_filsGauche"/>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Id fils droit:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="id_filsDroit"/>
        </item>
        <item>
         <widget class="QPushButton" name="create_oper">
          <property name="text">
           <string>Create</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="Line" name="line_2">
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="lineWidth">
           <number>4</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Primitive:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="prim_type">
          <item>
           <property name="text">
            <string>Disk</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Polygon</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="nb_sides">
          <property name="minimum">
           <number>3</number>
          </property>
          <property name="value">
           <number>4</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="create_prim">
          <property name="text">
           <string>Create</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="Line" name="line_3">
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="lineWidth">
           <number>4</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Translation</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QSlider" name="translationX">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="minimum">
             <number>-500</number>
            </property>
            <property name="maximum">
             <number>500</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="dsb_tx">
            <property name="decimals">
             <number>1</number>
            </property>
            <property name="minimum">
             <double>-500.000000000000000</double>
            </property>
            <property name="maximum">
             <double>500.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QSlider" name="translationY">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="minimum">
             <number>-500</number>
            </property>
            <property name="maximum">
             <number>500</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="dsb_ty">
            <property name="decimals">
             <number>1</number>
            </property>
            <property name="minimum">
             <double>-500.000000000000000</double>
            </property>
            <property name="maximum">
             <double>500.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Rotation</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QSlider" name="rotation">
            <property name="minimum">
             <number>-180</number>
            </property>
            <property name="maximum">
             <number>180</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="dsb_Rot">
            <property name="decimals">
             <number>1</number>
            </property>
            <property name="minimum">
             <double>-180.000000000000000</double>
            </property>
            <property name="maximum">
             <double>180.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Scale</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QSlider" name="scaleX">
            <property name="minimum">
             <number>-1000</number>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="pageStep">
             <number>10</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="dsb_sx">
            <property name="readOnly">
             <bool>false</bool>
            </property>
            <property name="minimum">
             <double>0.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1000.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QSlider" name="scaleY">
            <property name="minimum">
             <number>-1000</number>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="dsb_sy">
            <property name="readOnly">
             <bool>false</bool>
            </property>
            <property name="minimum">
             <double>0.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1000.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="resetTransfo">
            <property name="text">
             <string>Reset</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="applyTransfo">
            <property name="text">
             <string>Apply</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1038</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuCSG">
    <property name="title">
     <string>CSG</string>
    </property>
    <addaction name="actionLoadCSG"/>
    <addaction name="actionSaveCSG"/>
    <addaction name="actionAppendCSG"/>
    <addaction name="actionClearCSG"/>
   </widget>
   <widget class="QMenu" name="menuImage">
    <property name="title">
     <string>Image</string>
    </property>
    <addaction name="actionLoadIMG"/>
    <addaction name="actionSaveIMG"/>
    <addaction name="separator"/>
    <addaction name="actionDrawSobel"/>
   </widget>
   <widget class="QMenu" name="menuCurNode">
    <property name="title">
     <string>CurrentNode</string>
    </property>
    <addaction name="actionSwapLRRoot"/>
    <addaction name="actionUnjoinRoot"/>
    <addaction name="actionDraw_BB"/>
    <addaction name="actionClone"/>
   </widget>
   <addaction name="menuCSG"/>
   <addaction name="menuImage"/>
   <addaction name="menuCurNode"/>
  </widget>
  <action name="actionLoadCSG">
   <property name="text">
    <string>Load</string>
   </property>
  </action>
  <action name="actionSaveCSG">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionLoadIMG">
   <property name="text">
    <string>Load</string>
   </property>
  </action>
  <action name="actionSaveIMG">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionAppendCSG">
   <property name="text">
    <string>Append</string>
   </property>
  </action>
  <action name="actionClearCSG">
   <property name="text">
    <string>Clear</string>
   </property>
  </action>
  <action name="actionSwapLRRoot">
   <property name="text">
    <string>swapLR</string>
   </property>
  </action>
  <action name="actionUnjoinRoot">
   <property name="text">
    <string>unjoinRoot</string>
   </property>
  </action>
  <action name="actionDrawSobel">
   <property name="text">
    <string>drawSobel</string>
   </property>
  </action>
  <action name="actionDraw_BB">
   <property name="text">
    <string>draw BB</string>
   </property>
  </action>
  <action name="actionClone">
   <property name="text">
    <string>clone</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
